### PR TITLE
NXP-27649 blob computation temp dir delete on exit

### PR DESCRIPTION
**Pull request by andreinechaev**

### CODE REVIEW

File: nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/AbstractTransientBlobComputation.java [hunk 1]
```diff
@@ -47,6 +47,7 @@ public void init(ComputationContext context) {
         super.init(context);
         try {
             temp = Files.createTempDirectory(metadata().name());
+            temp.toFile().deleteOnExit();
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/AbstractTransientBlobComputation.java]: It is probably better to use `Framework#trackFile`.

Comment (andreinechaev) [reply nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/AbstractTransientBlobComputation.java]: thought about it but then what to use as ref? `context`? don't think the object will be alive needed time. 
The main complain was that temp folder wasn't cleaned up at server startup

Comment (kevinleturc) [reply nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/AbstractTransientBlobComputation.java]: We use the file itself as marker.

Comment (andreinechaev) [reply nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/AbstractTransientBlobComputation.java]: ok, thx

**Comment by ataillefer** [comment]
Would it be possible to add a unit test?

**Comment by andreinechaev** [comment]
> Would it be possible to add a unit test?

@ataillefer I don't see how we can test shutdown of jvm and cleanup of FS. open for suggestions.

**Comment by ataillefer** [comment]
I would've imagined that at some point the blob is not supposed to be referenced anymore (before shutdown), thus garbage collected.
But I don't know when this happens and if'ts possible to simulate.
Maybe it's not doable!

**Comment by kevinleturc** [comment]
It's probably possible to shutdown the computation runner which will free the computation and thus the Path. But this test will probably be random.

The marker I suggested may not be relevant, as `toFile` will return a new File, it will probably be garbage collected before the Path itself. Maybe interesting to use the Path as marker.

@andreinechaev can you squash your commits and change the marker? Next, I think a T&P could highlights issues if marker is not the right one.

**Comment by kevinleturc** [comment]
T&P is ok: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-anechaev/12/

**Pull request merged**